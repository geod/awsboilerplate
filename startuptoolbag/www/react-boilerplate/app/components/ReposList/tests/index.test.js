import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { render } from 'react-testing-library';

import ReposList from '../index';
import configureStore from '../../../configureStore';

describe('<ReposList />', () => {
  it('should render the loading indicator when its loading', () => {
    const { container } = render(<ReposList loading />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an error if loading failed', () => {
    const { queryByText } = render(
      <IntlProvider locale="en">
        <ReposList loading={false} error={{ message: 'Loading failed!' }} />
      </IntlProvider>,
    );
    expect(queryByText(/Something went wrong/)).not.toBeNull();
  });

  it('should render the repositories if loading was successful', () => {
    const store = configureStore(
      { global: { currentUser: 'mxstbr' } },
      browserHistory,
    );
    const repos = [
      {
        owner: {
          login: 'mxstbr',
        },
        html_url: 'https://github.com/react-boilerplate/react-boilerplate',
        name: 'react-boilerplate',
        open_issues_count: 20,
        full_name: 'react-boilerplate/react-boilerplate',
      },
    ];
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ReposList repos={repos} error={false} />
        </IntlProvider>
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render anything if nothing interesting is provided', () => {
    const { container } = render(
      <ReposList repos={false} error={false} loading={false} />,
    );

    expect(container.firstChild).toBeNull();
  });
});
