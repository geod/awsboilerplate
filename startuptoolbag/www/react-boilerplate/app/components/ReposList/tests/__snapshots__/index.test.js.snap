// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReposList /> should render the loading indicator when its loading 1`] = `
<div
  class="Wrapper-sc-1umgotm-0 laCwDT"
>
  <ul
    class="Ul-mwnq6h-0 jcxXFy"
  >
    <div
      class="Wrapper-sc-12uw37d-0 bTWIVF"
    >
      <div
        class="Circle__CirclePrimitive-ww56dy-0 enQIMS"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jCxmwS"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 dyFXB"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 ikwUeP"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 hwlQRz"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jjXxyb"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 jVExkJ"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 haCVIz"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 ghhiLZ"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 bCbqdt"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 hZPQMA"
      />
      <div
        class="Circle__CirclePrimitive-ww56dy-0 fsGOmP"
      />
    </div>
  </ul>
</div>
`;

exports[`<ReposList /> should render the repositories if loading was successful 1`] = `
<div
  class="Wrapper-sc-1umgotm-0 laCwDT"
>
  <ul
    class="Ul-mwnq6h-0 jcxXFy"
  >
    <li
      class="Wrapper-euo0oy-0 jLvxtc"
    >
      <div
        class="Item-sc-3y9mie-0 fKXSkT"
      >
        <div
          class="Wrapper-sc-17s0rao-0 dlZQhZ"
        >
          <a
            class="A-br8h0y-0 RepoLink-pvpwpn-0 eEveIZ"
            href="https://github.com/react-boilerplate/react-boilerplate"
            target="_blank"
          >
            react-boilerplate
          </a>
          <a
            class="A-br8h0y-0 IssueLink-uyzonb-0 kUUUnc"
            href="https://github.com/react-boilerplate/react-boilerplate/issues"
            target="_blank"
          >
            <svg
              class="IssueIcon-s8m34y-0 jEBeEF"
              height="1em"
              width="0.875em"
            >
              <path
                d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7S10.14 13.7 7 13.7 1.3 11.14 1.3 8s2.56-5.7 5.7-5.7m0-1.3C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7S10.86 1 7 1z m1 3H6v5h2V4z m0 6H6v2h2V10z"
              />
            </svg>
            <span>
              20
            </span>
          </a>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
